### PR TITLE
[Serializer] Fixed docblocks and parameter names

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32179
| License       | MIT
| Doc PR        | N/A

Backports from #33185.

### CODE REVIEW

File: src/Symfony/Component/Serializer/Serializer.php [hunk 1]
```diff
@@ -172,17 +172,17 @@ public function normalize($data, $format = null, array $context = [])
      *
      * @throws NotNormalizableValueException
      */
-    public function denormalize($data, $type, $format = null, array $context = [])
+    public function denormalize($data, $class, $format = null, array $context = [])
```
Comment (derrabus) [inline src/Symfony/Component/Serializer/Serializer.php]: 💡 The parameter is called `$class` in the interface. This change synchronizes the method signatures.

**Review by dunglas — APPROVED** [review]

**Comment by dunglas** [comment]
For the record, `$class`/`$type` can be other kind of types, including interfaces: https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php#L31

However, as even interfaces define the constant `::class` in PHP, using `$class` looks good enough for me.

**Comment by derrabus** [comment]
@dunglas We can also change `$type` to `$class` everywhere. My concern was just to sync the parameters. In case of the `ArrayDenoramalizer` we accept `SomeClass[]`, so `$type` might be more accurate.

**Comment by dunglas** [comment]
Indeed, maybe should we use `$type` everywhere then.

**Comment by derrabus** [comment]
Done.

**Review by maxhelias — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**